service/migration_manager.cc

```diff
@@ -746,7 +748,24 @@ future<std::vector<mutation>> prepare_column_family_update_announcement(storage_
         }
         co_await seastar::async([&] {
             db.get_notifier().before_update_column_family(*cfm, *old_schema, mutations, ts);
+
+            // Notify drop indices
+            auto diff = difference(old_schema->all_indices(), cfm->all_indices());
+
+            // indices that are no longer needed
+            for (auto&& name : diff.entries_only_on_left) {
+                const index_metadata& index = old_schema->all_indices().at(name);
+                schema_ptr view;
+                try {
+                    view = db.find_schema(old_schema->ks_name(), name + "_index");
+                } catch (...) {
+                    mlogger.error("Could not find schema for index {}.{} id={}", old_schema->ks_name(), name + "_index", index.id());
+                    continue;
+                }
+                db.get_notifier().before_drop_column_family(*view, mutations, ts);
```

Comment:
Do we have more asymmetries between tables and views?

Reply:
you mean views and indices?

Reply:
@avikivity what do you mean "asymmetries"?

As I noted in the past, the root of the need for this patch (and commits https://github.com/scylladb/scylladb/commit/88a5ddabcec625434b2df0ed82a63c36245af35f and https://github.com/scylladb/scylladb/commit/8181e287312c2b3eefa222eb6946cdae29195aa5 where I added the tablets for views and indexes) is the idea that you need "callbacks" to create and delete tablets - and you need to remember to call those callbacks in multiple places. Perhaps a more reliable approach would have been to have just one single function that can **create** a table, and the code that created a table, view, or index would all have had to call this single function - and then we would make this single function create the tablets. But this is not what we have today - we need to remember to call the callbacks from any code which can create or delete a table, even if this "table" is a view (or a view backing an index).

Reply:
asymmetry - callback called on table but not on view

Reply:
It's impossible to know :-( The "notifier" architecture makes it possible to create a table mutation without calling the notifier. I thought that last time I tracked all the missing places but it turns out I only fixed the view **creation** and now Benny discovered I forgot the view **deletion**. I don't know if there are more places like this. Testing should discover all these bugs (I discovered the missing creation notifiers call in the very first tests that I wrote for views and for secondary index).